### PR TITLE
Fix handling of zfs datasets and mount points with spaces

### PR DESCRIPTION
**Pull request by 3add3287**
### What does this PR do?
Properly quote the names of zfs datasets as well as mount points where necessary to handle cases where the path contains spaces

### What issues does this PR fix or reference?
#44401

### Previous Behavior
When paths for zfs filesystems/datasets or mount points had a space in them, the zfs command would be executed with the wrong arguments and throw an error.

### New Behavior
Dataset names or mount points are now properly quoted so spaces work as they should

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

**Review by cachedout — CHANGES_REQUESTED** [review]
We need to start by fixing up some lint errors which can be seen here:

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/16132/violations/file/salt/modules/zfs.py/

**Comment by rallytime** [comment]
Hi @3add3287 - There is a merge conflict here, as well as some test failures. Can you take a look?

**Comment by cachedout** [comment]
@sjorge Would you like to have a look at this?

File: salt/modules/zfs.py [hunk 1]
```diff
@@ -1261,6 +1275,7 @@ def get(*dataset, **kwargs):
     cmd = '{0} {1}'.format(cmd, properties)
 
     # datasets
+    dataset = [_zfs_quote_escape_path(x) for x in dataset]
```
Comment (sjorge) [inline salt/modules/zfs.py]: This looks a bit odd, does this work when multiple datasets are passed?

**Comment by sjorge** [comment]
@rallytime yeah this will also need a rebase, as the zfs grains change had changed the virtual function.

This will also effect #44371, if this goes in after this one that one will need a rebase too. If it goes in before it won't as I ready re-based.

Both will need to be in before I start work on #44404 as that will probably be a big change.

**Review by sjorge — APPROVED** [review]

**Comment by cachedout** [comment]
@3add3287 Can you please rebase this? It has merge conflicts.

**Comment by cachedout** [comment]
Bump @3add3287. We need the merge conflicts fixed before we can merge this, please.

**Comment by 3add3287** [comment]
Traveling, should have a chance to look at it tomorrow evening

**Comment by 3add3287** [comment]
Conflict seems easy to solve. Not sure I did it right in terms of git and it's workflow (more used to the likes of svn, perforce or, in the dark ages, cvs...)

**Review by sjorge — APPROVED** [review]

**Comment by rallytime** [comment]
@3add3287 It now appears that this PR is failing some related tests. Can you take a look?

**Comment by 3add3287** [comment]
I'm stuck in git hell and can't seem to get my branch properly updated with the diverging branches and pushing the commit. Spending more time reading git docs didn't help, nor did any the approaches I found via search work.
Anyone got a problem if I close this one and create a new PR ?

**Comment by 3add3287** [comment]
ping?

**Comment by rallytime** [comment]
@3add3287 Ah, sorry I didn't see your question until just now. That is totally fine if you want to create a new PR. Just link it here and close this one out.

**Comment by rallytime** [comment]
For the record, the tests that are failing are the following:

- unit.modules.test_zfs.ZfsTestCase.test_create_error_missing_dataset
- unit.modules.test_zfs.ZfsTestCase.test_create_error_missing_parent
- unit.modules.test_zfs.ZfsTestCase.test_create_error_no_such_pool
- unit.modules.test_zfs.ZfsTestCase.test_create_error_trailing_slash
- unit.modules.test_zfs.ZfsTestCase.test_create_success
- unit.modules.test_zfs.ZfsTestCase.test_create_success_with_create_parent

**Comment by rallytime** [comment]
@3add3287 Just a bump on this one. :)

**Comment by 3add3287** [comment]
Replaced by PR #45172

**Pull request closed without merging**